apps/pkt-gen/pkt-gen.c

```diff
@@ -1140,15 +1202,22 @@ send_packets(struct netmap_ring *ring, struct pkt *pkt, void *frame,
 	for (fcnt = nfrags, sent = 0; sent < count; sent++) {
 		struct netmap_slot *slot = &ring->slot[cur];
 		char *p = NETMAP_BUF(ring, slot->buf_idx);
-		int buf_changed = slot->flags & NS_BUF_CHANGED;
+		int copy = options & OPT_COPY || slot->flags & NS_BUF_CHANGED;
+
+		/* sender_body() drops OPT_COPY after starting, but we need
+		 * to copy over any ARP packets lingering in the txring */
+		if (!copy && slot->len == arp_size) {
```

Comment:
Could you replace this condition with `options & OPT_GARP`? In this way you make only a single check in the common case.